### PR TITLE
LPS-163464 [Raylife AP] Creation of the flow to the rejected status flag screen

### PR DESCRIPTION
**Pull request by eliasantosf**
Task: [LPS-163464](https://issues.liferay.com/browse/LPS-163464)

### CODE REVIEW

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/NewApplication/NewApplicationAuto/index.tsx [hunk 1]
```diff
@@ -112,17 +102,55 @@ const NewApplicationAuto = ({children}: DriverInfoProps) => {
...
+			createOrUpdateRaylifeApplication(state, applicationStatus).then(
+				(response) => {
+					const {
+						data: {id},
+					} = response;
+
+					dispatch({payload: {id}, type: ACTIONS.SET_APPLICATION_ID});
+				}
+			);
+		}
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/NewApplication/NewApplicationAuto/index.tsx:136]: Please, remove the else condition.

```suggestion
		if (
			state.currentStep === 2 &&
			form[0]?.accidentCitation[0]?.value ===
				'Citation - Driving under the influence'
		) {
			createOrUpdateRaylifeApplication(
				state,
				CONSTANTS.APPLICATION_STATUS.REJECTED
			).then((response) => {
				const {
					data: {id},
				} = response;

				dispatch({payload: {id}, type: ACTIONS.SET_APPLICATION_ID});
			});
			
			return dispatch({
				payload: 5,
				type: ACTIONS.SET_CURRENT_STEP,
			});
		}
		
		createOrUpdateRaylifeApplication(state, applicationStatus).then(
			(response) => {
				const {
					data: {id},
				} = response;

				dispatch({payload: {id}, type: ACTIONS.SET_APPLICATION_ID});
			}
		);
		
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/NewApplication/NewApplicationAuto/index.tsx [hunk 2]
```diff
@@ -112,17 +102,55 @@ const NewApplicationAuto = ({children}: DriverInfoProps) => {
...
 				payload: state.currentStep - 1,
 				type: ACTIONS.SET_CURRENT_STEP,
 			});
 		}
+		else {
+			dispatch({
+				payload: 2,
+				type: ACTIONS.SET_CURRENT_STEP,
+			});
+		}
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/NewApplication/NewApplicationAuto/index.tsx:153]: Please, remove the else condition.
```suggestion
		if (state.currentStep < 5) {
			return dispatch({
				payload: state.currentStep - 1,
				type: ACTIONS.SET_CURRENT_STEP,
			});
		}
		
		dispatch({
			payload: 2,
			type: ACTIONS.SET_CURRENT_STEP,
		});
		
```

**Review by joanagrego — APPROVED** [review]
Nice job! :white_check_mark:

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/NewApplication/NewApplicationAuto/index.tsx [hunk 3]
```diff
@@ -112,17 +102,55 @@ const NewApplicationAuto = ({children}: DriverInfoProps) => {
...
+				state,
+				CONSTANTS.APPLICATION_STATUS.REJECTED
+			).then((response) => {
+				const {
+					data: {id},
+				} = response;
+
+				dispatch({payload: {id}, type: ACTIONS.SET_APPLICATION_ID});
+			});
+			dispatch({
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/NewApplication/NewApplicationAuto/index.tsx:121]: Please add a blank line between line 120 and 121.

**Review by silasbsi — CHANGES_REQUESTED** [review]
I added some suggestions.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/NewApplication/NewApplicationAuto/index.tsx (hunk 1 above)
Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/NewApplication/NewApplicationAuto/index.tsx]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/NewApplication/NewApplicationAuto/index.tsx (hunk 2 above)
Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/NewApplication/NewApplicationAuto/index.tsx]: Done!

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Review by joanagrego — APPROVED** [review]
Nice job! :white_check_mark:

**Review by silasbsi — APPROVED** [review]
LGMT :+1: Thanks @eliasantosf

**Comment by silasbsi** [comment]
ci:forward

**Comment by eliasantosf** [comment]
ci:forward

**Pull request closed without merging**